File: Iteration_2/src/main/java/NonTechnicalElectiveCourse.java
public class NonTechnicalElectiveCourse extends Course {

    public NonTechnicalElectiveCourse(String courseId, String courseName, int credits, Course prerequisiteCourse) {
        super(courseId, courseName, credits, prerequisiteCourse);
    }
    public NonTechnicalElectiveCourse(String courseId, String courseName, String grade, int credits) {
        super(courseId, courseName, grade, credits);
    }
    public NonTechnicalElectiveCourse(String courseId, String courseName, int credits) {
        super(courseId, courseName, credits);
    }

    @Override
    public String getCourseType() {
        return "Non-Technical Elective";
    }
}

File: Iteration_3/NonTechnicalElectiveCourse.py
from Course import Course

class NonTechnicalElectiveCourse(Course):
    def __init__(self, course_id: str, course_name: str, credits: int, prerequisite_course: 'Course' = None):
        if prerequisite_course:
            super().__init__(course_id, course_name, credits, prerequisite_course)
        else:
            super().__init__(course_id, course_name, credits)



    def get_course_type(self):
        return "Non-Technical Elective"
